common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -343,9 +343,23 @@ public MicrosoftStsTokenRequest createTokenRequest(@NonNull final MicrosoftStsAu
         tokenRequest.setClientId(request.getClientId());
         tokenRequest.setScope(request.getTokenScope());
         tokenRequest.setClaims(request.getClaims());
-        tokenRequest.setGrantType(TokenRequest.GrantTypes.AUTHORIZATION_CODE);
         setTokenRequestCorrelationId(tokenRequest);
 
+        // Existence of a device code inside of the response object implies Device Code Flow is being used
+        if (response.getDeviceCode() != null) {
+            tokenRequest.setGrantType(TokenRequest.GrantTypes.DEVICE_CODE);
+            tokenRequest.setDeviceCode(response.getDeviceCode());
+
+            // mTenantId from mAudience (being "common") is being added to the path in AzureActiveDirectoryAuthority
+            // Needed to omit "\common" from the endpoint so as to not break DCF
+            // mTokenEndPoint be automatically reset in future commands
+            String newTokenEndpoint = this.mTokenEndpoint.replace("/common/", "/");
```

Comment:
We shouldn't be doing this hack -- we need to find the exact issue here

Reply:
Fixed